File: Q0257_Binary_Tree_Paths/Solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def binaryTreePaths(self, root: TreeNode):
        res = []
        if root is None:
            return res

        if root.left is None and root.right is None:
            res.append(str(root.val))
            return res
        left_res = self.binaryTreePaths(root.left)
        for i in left_res:
            res.append(str(root.val) + "->" + i)
        right_res = self.binaryTreePaths(root.right)
        for i in right_res:
            res.append(str(root.val) + "->" + i)
        return res

File: Q0257_Binary_Tree_Paths/Solution.java
package Q0257_Binary_Tree_Paths;

import java.util.ArrayList;
import java.util.List;

/**
 * @ Description:
 * @ Date: Created in 16:12 2019-04-13
 * @ Author: Anthony_Duan
 */
public class Solution {
    class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    public List<String> binaryTreePaths(TreeNode root) {

        List<String> res = new ArrayList<>();

        if (root == null) {
            return res;
        }
        if (root.left == null && root.right == null) {
            res.add(String.valueOf(root.val));
            return res;
        }

        List<String> ls = binaryTreePaths(root.left);
        for (String l : ls) {
            res.add(root.val + "->" + l);
        }
        List<String> rs = binaryTreePaths(root.right);
        for (String r : rs) {
            res.add(root.val + "->" + r);
        }

        return res;

    }
}
